src/php/ext/grpc/channel.c

```diff
@@ -518,20 +603,137 @@ static void php_grpc_channel_plink_dtor(php_grpc_zend_resource *rsrc
     return;
   }
   if (le->channel != NULL) {
-    gpr_mu_lock(&le->channel->mu);
-    if (le->channel->wrapped != NULL) {
-      grpc_channel_destroy(le->channel->wrapped);
-      free(le->channel->args_hashstr);
-      le->channel->wrapped = NULL;
-      le->channel->target = NULL;
-      le->channel->args_hashstr = NULL;
-      free(le->channel->key);
-      le->channel->key = NULL;
-    }
-    gpr_mu_unlock(&le->channel->mu);
+    php_grpc_channel_unref(le->channel);
+    le->channel = NULL;
+  }
+  free(le);
+  le = NULL;
+}
+
+// A destructor associated with each list entry from the target_bound map
+static void php_grpc_target_bound_dtor(php_grpc_zend_resource *rsrc
+                                        TSRMLS_DC) {
+  target_bound_le_t *le = (target_bound_le_t *) rsrc->ptr;
+  if (le == NULL) {
+    return;
   }
+  free(le);
+  le = NULL;
+}
+
+#ifdef GRPC_PHP_DEBUG
```

Comment:
It will only be compiled when the user set `enable-tests`:

`./configure --enable-tests`